src/python/foglamp/configuration_manager.py

```diff
@@ -0,0 +1,415 @@
+
+    Exceptions Raised:
+    ValueError
+    TypeError
+
+    Restrictions and Usage:
+    A FogLAMP component calls this method to create one or more new configuration categories to store initial configuration.
+    Only default values can be entered for and item's entries.
+    A "value" entry specified for an item will raise an exception.
+    """
+    category_val_prepared = ''
+    try:
+        # validate new category_val
+        category_val_prepared = await _validate_category_val(category_value, True)
+        # check if category_name is already in storage
+        category_val_storage = await _read_category_val(category_name)
+        if category_val_storage is None:
+            return await _create_new_category(category_name, category_val_prepared, category_description)
+        else:
+            # validate category_val from storage
+            try:
+                category_val_storage = await _validate_category_val(category_val_storage, False)
+            except:
+                _logger.exception(
+                    'category_value for category_name {} from storage is corrupted; using category_value without merge'.format(
+                        category_name))
+            else:
+                category_val_prepared = await _merge_category_vals(category_val_prepared, category_val_storage)
+            return await _update_category(category_name, category_val_prepared, category_description)
+
+    except:
+        _logger.exception(
+            'Unable to create new category based on category_name {} and category_description {} and category_json_schema {}'.format(
+                category_name, category_description, category_val_prepared))
+        raise
+    return None
+
+def register_category(category_name, callback):
+    pass
+
+# async def main_test():
+#     sample_json = {
+#         "port": {
+#             "description": "Port to listen on",
+#             "default": "5432",
+#             "type": "integer"
+#         },
+#         "url": {
+#             "description": "URL to accept data on",
+#             "default": "sensor/reading-values",
+#             "type": "string"
+#         },
+#         "certificate": {
+#             "description": "X509 certificate used to identify ingress interface",
+#             "default": "47676565",
+#             "type": "X509 certificate"
+#         }
+#     }
+#
+#     print("test create_category")
+#     await create_category('CATEG', sample_json, 'CATEG_DESCRIPTION')
+#
+#     print("test get_all_category_names")
+#     rows = await get_all_category_names()
+#     for row in rows:
+#         print(row)
+#
+#     print("test get_all_items")
+#     json = await get_all_items('CATEG')
+#     print(json)
+#     print(type(json))
+#
+#     print("test get_item")
+#     json = await get_item('CATEG', "url")
+#     print(json)
+#     print(type(json))
+#
+#     print("test get_value")
+#     json = await get_value('CATEG', "url")
+#     print(json)
```

Comment:
return is not json but just string. for all get_value

you may want to store in `result` as you are printing and doing type checking

> Understood that its test code only. you may want to add assert type(json) == dict or assert type(json) == str instead of print